include/pybind11/pytypes.h

```diff
@@ -2057,6 +2073,10 @@ iterator object_api<D>::end() const {
     return iterator::sentinel();
 }
 template <typename D>
+item_accessor object_api<D>::operator[](object &&key) const {
```

Comment:
The declaration appears after the overload for `handle`. Could you please put them in the same order here?